mod/pages/start.php

```diff
@@ -382,6 +378,16 @@ function pages_container_permission_check($hook, $type, $return, $params) {
 		return;
 	}
 
+	$parent_container = $container->getContainerEntity();
+	if ($parent_container instanceof ElggGroup) {
+		// Only group members should be allowed to create subpages
+		return $parent_container->canWriteToContainer();
+	}
+
+	if (!elgg_in_context('pages')) {
```

Comment:
This logic is really bad and will fail during actions or in case of routed request.